apps/theming/lib/Jobs/RestoreBackgroundImageColor.php

```diff
@@ -0,0 +1,206 @@
+ */
+
+namespace OCA\Theming\Jobs;
+
+use OCA\Theming\AppInfo\Application;
+use OCA\Theming\Service\BackgroundService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\BackgroundJob\QueuedJob;
+use OCP\Files\IAppData;
+use OCP\Files\NotFoundException;
+use OCP\Files\NotPermittedException;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use Psr\Log\LoggerInterface;
+
+class RestoreBackgroundImageColor extends QueuedJob {
+	public const TIME_SENSITIVE = 0;
+
+	public const STAGE_PREPARE = 'prepare';
+	public const STAGE_EXECUTE = 'execute';
+	// will be saved in appdata/theming/global/
+	protected const STATE_FILE_NAME = '30_background_image_color_restoration.json';
+
+	public function __construct(
+		ITimeFactory $time,
+		private IConfig $config,
+		private IAppData $appData,
+		private IJobList $jobList,
+		private IDBConnection $dbc,
+		private LoggerInterface $logger,
+		private BackgroundService $service,
+	) {
+		parent::__construct($time);
+	}
+
+	protected function run(mixed $argument): void {
+		if (!is_array($argument) || !isset($argument['stage'])) {
+			throw new \Exception('Job '.self::class.' called with wrong argument');
+		}
+
+		switch ($argument['stage']) {
+			case self::STAGE_PREPARE:
+				$this->runPreparation();
+				break;
+			case self::STAGE_EXECUTE:
+				$this->runMigration();
+				break;
+			default:
+				break;
+		}
+	}
+
+	protected function runPreparation(): void {
+		try {
+			$qb = $this->dbc->getQueryBuilder();
+			$qb2 = $this->dbc->getQueryBuilder();
+
+			$innerSQL = $qb2->select('userid')
+				->from('preferences')
+				->where($qb2->expr()->eq('configkey', $qb->createNamedParameter('background_color')));
+
+			// Get those users, that have a background_image set - not the default, but no background_color.
+			$result = $qb->selectDistinct('a.userid')
+				->from('preferences', 'a')
+				->leftJoin('a', $qb->createFunction('('.$innerSQL->getSQL().')'), 'b', 'a.userid = b.userid')
+				->where($qb2->expr()->eq('a.configkey', $qb->createNamedParameter('background_image')))
+				->andWhere($qb2->expr()->neq('a.configvalue', $qb->createNamedParameter(BackgroundService::BACKGROUND_DEFAULT)))
+				->andWhere($qb2->expr()->isNull('b.userid'))
+				->executeQuery();
+
+			$userIds = $result->fetchAll(\PDO::FETCH_COLUMN);
+			$this->logger->info('Prepare to restore background information for {users} users', ['users' => count($userIds)]);
+			$this->storeUserIdsToProcess($userIds);
+		} catch (\Throwable $t) {
+			$this->jobList->add(self::class, ['stage' => self::STAGE_PREPARE]);
+			throw $t;
+		}
+		$this->jobList->add(self::class, ['stage' => self::STAGE_EXECUTE]);
+	}
+
+	/**
+	 * @throws NotPermittedException
+	 * @throws NotFoundException
+	 */
+	protected function runMigration(): void {
+		$allUserIds = $this->readUserIdsToProcess();
+		$notSoFastMode = count($allUserIds) > 1000;
+
+		$userIds = $notSoFastMode ? array_slice($allUserIds, 0, 1000) : $allUserIds;
```

Comment:
```suggestion
		$userIds = array_slice($allUserIds, 0, 1000);
```
No need to check count first, `array_slice` will take care of it.